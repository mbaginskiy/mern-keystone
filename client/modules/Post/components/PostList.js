import React from 'react';
import PropTypes from 'prop-types';

// Import Components
import Post from './Post';

const PostList = ({ posts, handleDeletePost }) => (
  <div className="listView">
    {
      posts.map(post => (
        <Post
          post={post}
          key={post.id}
          type="brief"
          onDelete={() => handleDeletePost(post.id)}
        />
      ))
    }
  </div>
);

PostList.propTypes = {
  posts: PropTypes.arrayOf(PropTypes.shape({
    author: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    content: PropTypes.shape({
      brief: PropTypes.string.isRequired,
      extended: PropTypes.string.isRequired,
    }).isRequired,
    slug: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
  })).isRequired,
  handleDeletePost: PropTypes.func.isRequired,
};

export default PostList;
