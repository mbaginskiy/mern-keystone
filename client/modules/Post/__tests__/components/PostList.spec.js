import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import PostList from '../../components/PostList';

const posts = [
  {
    author: 'Admin User',
    title: 'Test Title 1',
    slug: 'test-title-1',
    content: {
      brief: 'test',
      extended: 'test',
    },
    id: '1',
  },
  {
    author: 'Admin User',
    title: 'Test Title 2',
    slug: 'test-title-2',
    content: {
      brief: 'test',
      extended: 'test',
    },
    id: '2',
  },
];

test('renders the list', () => {
  const wrapper = shallow(
    <PostList
      posts={posts}
      handleShowPost={() => {}}
      handleDeletePost={() => {}}
    />,
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('Post').length).toBe(2);
});
