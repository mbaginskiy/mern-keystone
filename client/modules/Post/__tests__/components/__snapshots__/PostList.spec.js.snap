// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the list 1`] = `
<div
  className="listView"
>
  <Post
    key="1"
    onDelete={[Function]}
    post={
      Object {
        "author": "Admin User",
        "content": Object {
          "brief": "test",
          "extended": "test",
        },
        "id": "1",
        "slug": "test-title-1",
        "title": "Test Title 1",
      }
    }
    type="brief"
  />
  <Post
    key="2"
    onDelete={[Function]}
    post={
      Object {
        "author": "Admin User",
        "content": Object {
          "brief": "test",
          "extended": "test",
        },
        "id": "2",
        "slug": "test-title-2",
        "title": "Test Title 2",
      }
    }
    type="brief"
  />
</div>
`;
